### ISSUE TITLE
Image processing: write text over image

### ISSUE THREAD

**Issue by h4tr3d** [gohugoio/hugo#6731]
Hi guys!

My idea: add ability to write text over image during processing.

It can allow generating default post cover images, when user does not point it manually.

Something like: https://habr.com/share/publication/473548/1aa369f9a9242124c031f7f05c13eee7/?v=1 for https://habr.com/en/company/encry/blog/473548/

**Comment by bep** [comment]
We have another issue about an `Overlay` method.

I don't remember the signature discussed there, but I guess something ala:

```
$img := $img.Overlay $anotherImage
```

With that in mind, if we create a function that creates some image from text, it could maybe look something like this:

```
{{ $background := resources.Get "clouds.jpg" }}
{{ $text := images.Print (dict "text" "Hello World") }}
{{ $img := $text.Overlay $background }}
```

Or something.

**Comment by h4tr3d** [comment]
Yep, it is applicable. But, next possibilities should be present:
1. text color configuration
2. BG color configuration
3. BG transparency level configuration
4. Text font (?)
5. Limit text image width/height

And, maybe, font border and inner color separately.

**Comment by frerich** [comment]
I suppose something akin to `image.Overlay` could be useful, but I'm skeptical about `image.Print`: there is a plethora of options to consider (sizes, colors, alignment, fonts etc.). Also, the browser (via CSS) is fairly good at rendering fonts (including consideration of aforementioned attributes) already. I.e. `images.Print` would duplication functionality already available in the browser.

So maybe it would be better to have a shortcode example which, given some image and some text (+ options) yields HTML with inline CSS to render the text on top of the image?

**Comment by bep** [comment]
@frerich @h4tr3d the only way this goes into Hugo is if we can make it simple, which would mean very basic and automatic text formatting and font handling etc.

The obvious use case for this is to get some auto-generated images to use in Open Graph/Twitter tags.

I have wanted this for the Hugo docs site. There we pick up an image in the bundle if defined with a fallback to a site-wide image.

It would be really cool if we could have a generic Hugo background image and then print the page title on top of it.
